src/api/events/mutate/sendEvent.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  EventCreatedSchema,
+  EventCreatedSchemaType,
+  NewEventSchemaType,
+} from "../schema";
+
+import { apiFactory } from "~/api/apiFactory";
+import { useApiKey } from "~/util/store/apiKey";
+import { useMutation } from "@tanstack/react-query";
+import { useNamespace } from "~/util/store/namespace";
+import { useToast } from "~/design/Toast";
+import { useTranslation } from "react-i18next";
+
+export const sendEvent = apiFactory({
+  url: ({ baseUrl, namespace }: { baseUrl?: string; namespace: string }) =>
+    `${baseUrl ?? ""}/api/namespaces/${namespace}/broadcast`,
+  method: "POST",
+  schema: EventCreatedSchema,
+});
+
+export const useSendEvent = ({
+  onSuccess,
+}: {
+  onSuccess?: (data: EventCreatedSchemaType) => void;
+} = {}) => {
+  const apiKey = useApiKey();
+  const namespace = useNamespace();
+  const { toast } = useToast();
+  const { t } = useTranslation();
+
+  if (!namespace) {
+    throw new Error("namespace is undefined");
+  }
+
+  const mutationFn = (data: NewEventSchemaType) =>
+    sendEvent({
+      apiKey: apiKey ?? undefined,
+      payload: data.body,
+      headers: {
+        "content-type": "application/cloudevents+json",
+      },
+      urlParams: {
+        namespace,
+      },
+    });
+
+  return useMutation({
+    mutationFn,
+    // no cache invalidation because events are updated via streaming
```

Comment:
As discussed in person, I briefly tried out adding cache invalidation and decided against it because (a) we don't need it in the current use case (due to streaming) and (b) it is not trivial to implement (so the time investment isn't justified right now).

I updated the comment regarding this in efd8d9523b04af1a7b6d906d634bb65a3ed23c75